### ISSUE TITLE
Is there a plan to have a FP16 for GPU so to have larger batch size or longer text documents support ?

### ISSUE THREAD

**Issue by howardhsu** [huggingface/pytorch-pretrained-BERT#10]
Is there a plan to have an FP16 for GPU so to have a larger batch size or longer text documents support?

**Comment by thomwolf** [comment]
Yes probably. I am testing fp16 right now. If it works well I will push it to the repo.

**Comment by thomwolf** [comment]
Ok I've added FP16 support (see updated readme)

**Comment by howardhsu** [comment]
Thanks for this quick updates.

**Comment by Ashish-Gupta03** [comment]
I'm not able to work with FP16 for pytorch BERT code. Particularly for BertForSequenceClassification, which I tried and got the issue
**Runtime error: Expected scalar type object Half but got scalar type Float for argument #2 target**
when I enabled fp16.
Also when using 
`logits = logits.half()
labels = labels.half()`
then the epoch time also increased.
